examples2/src/counter_pack.rs

```diff
@@ -181,96 +182,14 @@ mod __counter_pack_wasm_parts {
-    use odra::{prelude::*, EntryPointsCaller};
-    use odra::{types::Address, CallDef, HostEnv};
-
-    use crate::counter_pack::CounterPack;
-
-    pub struct CounterPackHostRef {
-        address: Address,
-        env: HostEnv
-    }
-
-    impl CounterPackHostRef {
-        pub fn get_count(&self, index_a: u8, index_b: u8) -> u32 {
-            self.env
-                .call_contract(
-                    self.address,
-                    CallDef::new(
-                        String::from("get_count"),
-                        runtime_args! {
-                            "index_a" => index_a,
-                            "index_b" => index_b
-                        }
-                    )
-                )
-                .unwrap()
-        }
-
-        pub fn increment(&self, index_a: u8, index_b: u8) {
-            self.env
-                .call_contract(
-                    self.address,
-                    CallDef::new(
-                        String::from("increment"),
-                        runtime_args! {
-                            "index_a" => index_a,
-                            "index_b" => index_b
-                        }
-                    )
-                )
-                .unwrap()
-        }
-    }
-
-    pub struct CounterPackDeployer;
-
-    impl CounterPackDeployer {
-        pub fn init(env: &HostEnv) -> CounterPackHostRef {
-            let epc = EntryPointsCaller::new(env.clone(), |contract_env, call_def| {
-                use odra::types::ToBytes;
-                let mut counter_pack = CounterPack::new(Rc::new(contract_env));
-                match call_def.method() {
-                    "get_count" => {
-                        let index_a: u8 = call_def.get("index_a").unwrap();
-                        let index_b: u8 = call_def.get("index_b").unwrap();
-                        let result = counter_pack.get_count(index_a, index_b);
-                        Bytes::from(result.to_bytes().unwrap())
-                    }
-                    "increment" => {
-                        let index_a: u8 = call_def.get("index_a").unwrap();
-                        let index_b: u8 = call_def.get("index_b").unwrap();
-                        counter_pack.increment(index_a, index_b);
-                        Bytes::from(vec![])
-                    }
-                    _ => panic!("Unknown method")
-                }
-            });
-
-            let address = env.new_contract("counter_pack", None, Some(epc));
-
-            CounterPackHostRef {
-                address,
-                env: env.clone()
-            }
-        }
-    }
-}
-
-#[cfg(not(target_arch = "wasm32"))]
-pub use __counter_pack_test_parts::*;
-
 #[cfg(test)]
 mod tests {
     pub use super::*;
 
     #[test]
     fn counter_pack_works() {
         let env = odra::test_env();
-        let counter_pack = CounterPackDeployer::init(&env);
+        let mut counter_pack = CounterPackDeployer::init(&env);
```

Comment:
Why it needs to be mut?

Reply:
HostRefs functions have the same signature as module functions. If a module function is mutable, a ref function is mutable as well. Not required but easier to implement, may be changed in future.